app/main.py

```diff
@@ -1 +1,50 @@
-#  write your code here
+class Animal():
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self.name = name
+        self. appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self,) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = not self.is_hungry
+            return self. appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(
+            self,
+            name: str,
+            is_hungry: bool = True
+    ) -> None:
+        super().__init__(name, appetite=3, is_hungry=is_hungry)
+
+    def catch_mouse(self) -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(
+            self,
+            name: str,
+            is_hungry: bool = True
+    ) -> None:
+        super().__init__(name, appetite=7, is_hungry=is_hungry)
+
+    def bring_slippers(self) -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list[Animal]) -> int:
+    food_points_needed = [animal.feed() for animal in animals
+                          if animal.is_hungry]
+    return sum(food_points_needed)
```

Comment:
Place the generator expression inside of the `sum`, there is no need to create a list via list comprehension here. In this way, you could omit to create the `food_points_needed` variable. Besides that, there is no need for the if condition here, the check is already performed in the `feed` method.